app/controllers/ops_controller/settings/upload.rb

```diff
@@ -115,4 +115,22 @@ def logo_dir
     Dir.mkdir(dir) unless dir.exist?
     dir.to_s
   end
+
+  def valid_file?(file, type)
+    if type == 'ico'
+      valid_magic_number = "00000100"
+      magic_number_length = 4
+    else
+      valid_magic_number = "89504e470d0a1a0a" # .png format
+      magic_number_length = 8
+    end
+
+    temp_filepath = file.tempfile.path
+    tempfile = File.new(temp_filepath, 'r')
+    tempfile.rewind
+    magic_number = tempfile.readpartial(magic_number_length).unpack("H*").first
+    file_extensions = file.original_filename.split(".")
```

Comment:
Separately I'm having trouble following this.  What type of object is `file`, and then what is `file.tempfile`?